tests/shared.py

```diff
@@ -204,11 +222,40 @@ def driver_missing_features(self, *features):
     def driver_supports_features(self, *features):
         return not self.driver_missing_features(*features)
 
+    def _bolt_version_to_feature(self, version):
+        if isinstance(version, protocol.Feature):
+            return self.driver_supports_features(version)
+        elif isinstance(version, str):
+            m = re.match(r"\D*(\d+)(?:\D+(\d+))?", version)
+            if not m:
+                raise ValueError("Invalid bolt version specidication")
+            version = tuple(map(int, m.groups("0")))
+        else:
+            version = tuple(version)
+        version = tuple(map(str, version))
+        if len(version) == 1:
+            version = (version[0], "0")
+        try:
+            return getattr(protocol.Feature, "_".join(("BOLT", *version)))
+        except AttributeError:
+            raise ValueError("Unknown bolt feature "
+                             + "_".join(("BOLT", *version)))
+
+    def driver_supports_bolt(self, version):
+        return self.driver_supports_features(
+            self._bolt_version_to_feature(version)
+        )
+
     def skip_if_missing_driver_features(self, *features):
         missing = self.driver_missing_features(*features)
         if missing:
             self.skipTest("Needs support for %s" % ", ".join(missing))
 
+    def skip_if_missing_bolt_support(self, version):
```

Comment:
Is this used somewhere?

Reply:
`skip_if_missing_driver_features` is/was used in exactly one place: in the `setUp` method of a base class to grammatically skip tests. Tbf. those methods are only useful in very specific cases. But I think it hurts little to have them there.